beetsplug/edit.py

```diff
@@ -313,8 +352,9 @@ def apply_data(self, objs, old_data, new_data):
             if forbidden:
                 continue
 
-            id = int(old_dict['id'])
-            apply(obj_by_id[id], new_dict)
+            # Reconcile back the user edits, using the reference_field.
+            val = self.obj_from_ref(old_dict)
```

Comment:
Hmm. As long as the IDs never actually get added to the database, I think they can be anything unique---so even just "i" should be a fine way to assign them. Then the IDs can be removed again before they objects get added to the database (at which point they'll be assigned real IDs).